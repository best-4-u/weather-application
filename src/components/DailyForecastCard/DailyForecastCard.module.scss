.daily_forecast_card {
  width: 100%;
  display: flex;
  flex-direction: column;
  padding: 1rem;
  background-color: white;
  border-radius: 4px;
  height: 22rem;

  > p {
    text-align: center;
  }

  > img {
    width: 10rem;
    height: 10rem;
    margin: auto;
  }

  .temp_footer {
    display: flex;
    justify-content: space-between;
    align-items: end;
    > div {
      > p {
        text-align: center;
        font-size: 0.9rem;
        margin-block-end: 0;

        > span::before {
          content: "\A";
          white-space: pre;
        }
      }
    }
  }
}
