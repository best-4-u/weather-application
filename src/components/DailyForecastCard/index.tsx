import React from "react";
import styles from "./DailyForecastCard.module.scss";
import { DailyWeatherCard } from "../../store/features/weather/types";

type Props = DailyWeatherCard;

function DailyForecastCard(props: Props): JSX.Element {

  const { date, min, max, weatherCode } = props; 

  return (
    <div className={styles.daily_forecast_card}>
      <p> { date } </p>
      <img src={`/${weatherCode}}`} alt="Weather icon" />
      <div className={styles.temp_footer}>
        <div>
          <p> min temp: <span> { min } &#8451;</span> </p>
        </div>
        <div>
          <p> max temp: <span> { max } &#8451;</span> </p>
        </div>
      </div>
    </div>
  );

}

export default DailyForecastCard;
