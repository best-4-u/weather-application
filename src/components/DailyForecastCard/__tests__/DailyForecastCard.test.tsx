import React from "react";
import renderer from "react-test-renderer";
import DailyForecastCard from "../index";

describe("DailyForecastCard", () => {
  it('default view', () => {
    const component = renderer.create(<DailyForecastCard min={0} max={0} date="1994-09-01" weatherCode={50} /> );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});


