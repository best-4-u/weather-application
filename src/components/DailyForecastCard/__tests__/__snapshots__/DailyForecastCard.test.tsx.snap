// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DailyForecastCard changes the class when hovered 1`] = `
<div
  className="daily_forecast_card"
>
  <p>
     
    01 September
     
  </p>
  <img
    alt="Weather icon"
    src="/rain.png"
  />
  <div
    className="temp_footer"
  >
    <div>
      <p>
        min temp: 
        <span>
           
          0
           ℃
        </span>
      </p>
    </div>
    <div>
      <p>
        max temp: 
        <span>
           
          0
           ℃
        </span>
      </p>
    </div>
  </div>
</div>
`;
